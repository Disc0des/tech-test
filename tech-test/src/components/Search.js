import React, { useState } from "react";
import getImages from "../requests/getImages";
import "../styles/Search.css";

const Search = ({ setSearchResults }) => {
  const [searchValue, setSearchValue] = useState();

  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults(await getImages(searchValue));
  };

  return (
    <div className="Search">
      <form className="seach-form" onSubmit={handleSubmit}>
        <input 
          className="search-input" 
          type="text"
          onChange={(e) => setSearchValue(e.target.value)}
        />
        <button
          className="search-button"
          type="submit"
        >Launch!</button>
      </form>
    </div>
  );
};

export default Search;