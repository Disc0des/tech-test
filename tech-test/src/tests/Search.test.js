import { render } from "@testing-library/react";
import React from "react";
import Search from "../components/Search";

describe("Search", () => {
  it("renders Search component correctly", () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });
});