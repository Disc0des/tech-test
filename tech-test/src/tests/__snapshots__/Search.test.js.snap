// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders Search component correctly 1`] = `
<DocumentFragment>
  <form
    class="seach-form"
  >
    <input
      class="search-input"
      type="text"
      value=""
    />
    <button
      class="search-button"
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
